### PR TITLE
Cumulative Update for Run2_UL Changes

### PR DESCRIPTION
**Pull request by JHiltbrand**
This PR will contain changes as a reaction to changes in `TreeMaker` for UL.

--removed `JetsAK8_prunedMass`, only prepared, never used in framework
--changed `MiniIso` to `iso` based on name change in `TreeMaker`
--can just simply get PU weights from `TreeMaker` which come from central ROOT files
--remove old BDT and Fisher files that are no longer used
--add templated `calcMT` and wrappers to `ROOT::Math::VectorUtil::DeltaR` and `ROOT::Math::VectorUtil::DeltaPhi` functions
--updates all relevant headers and source to play nicely with LorentzVector
--template some conversion functions to switch from collections of different LorentzVectors
--sort out how to get AK8 subjets and make nested vector of vectors again for TopTagger
--address type changes that have occurred with `TreeMaker`

### CODE REVIEW

File: Framework/include/PrepNTupleVars.h [hunk 1]
```diff
@@ -358,13 +354,12 @@ class PrepNTupleVars
         auto& newJetsAK8_NsubjettinessTau2  = tr.createDerivedVec<float>("JetsAK8"+name+"_NsubjettinessTau2", jc.JetsAK8.size());
         auto& newJetsAK8_NsubjettinessTau3  = tr.createDerivedVec<float>("JetsAK8"+name+"_NsubjettinessTau3", jc.JetsAK8.size());
         auto& newJetsAK8_softDropMass       = tr.createDerivedVec<float>("JetsAK8"+name+"_softDropMass", jc.JetsAK8.size());
-        auto& newJetsAK8_prunedMass         = tr.createDerivedVec<float>("JetsAK8"+name+"_prunedMass", jc.JetsAK8.size());
```
Comment (cmadrid1) [inline Framework/include/PrepNTupleVars.h:361]: Does the top tag not need the "prunedMass" variable?

File: Framework/include/ScaleFactors.h [hunk 2]
```diff
@@ -556,32 +553,18 @@ class ScaleFactors
         // For 2016 and 2018: Grab the individual pileup weight from the histogram found in PileupHistograms_*.root
         // For 2017: Using the puWeight stored in the nTuples
         // ----------------------------------------------------------------------------        
-        const auto& puWeight  = tr.getVar<double>("puWeight");
-        const auto& puSysUp   = tr.getVar<double>("puSysUp");
-        const auto& puSysDown = tr.getVar<double>("puSysDown");
-        const auto& tru_npv   = tr.getVar<double>("TrueNumInteractions");
+        const auto& puWeightUnCorr  = tr.getVar<double>("puWeight");
+        const auto& puSysUpUnCorr   = tr.getVar<double>("puSysUp");
+        const auto& puSysDownUnCorr = tr.getVar<double>("puSysDown");
```
Comment (cmadrid1) [inline Framework/include/ScaleFactors.h:558]: Are these variables stored in the new UL nTuples? I thought they made all new variables floats?

File: Framework/include/Utility.h [hunk 3]
```diff
@@ -23,6 +25,13 @@ namespace utility
     template<typename T> T sum2(T v) { return v*v; }
     template<typename T, typename... Args> T sum2(T v, Args... args) { return v*v + sum2(args...); }
     template<typename T, typename... Args> T addInQuad(T v, Args... args) { return sqrt(sum2(v, args...)); }
+    template<typename LV1, typename LV2>
+    double calcMT(const LV1& lepton, const LV2& met)
```
Comment (cmadrid1) [inline Framework/include/Utility.h:29]: Do we need to calculate MT somewhere? Also, I love the use of a template here.

File: Framework/include/PrepNTupleVars.h (hunk 1 above)
Comment (JHiltbrand) [reply Framework/include/PrepNTupleVars.h:361]: I could only find reference to it in `PrepNTupleVars.h`, no where to be seen in `RunTopTagger` or `SetupTopTagger`. Unless it is somehow implicitly passed.

File: Framework/include/Utility.h (hunk 3 above)
Comment (JHiltbrand) [reply Framework/include/Utility.h:29]: Special case reference in `Analyze1Lep` for calculating `mT` for use in the long ago W control region. Presumably reference to W control region could be removed.

File: Framework/include/ScaleFactors.h (hunk 2 above)
Comment (JHiltbrand) [reply Framework/include/ScaleFactors.h:558]: They are in, presumably stored as `float`

Comment (cmadrid1) [reply Framework/include/ScaleFactors.h:558]: Ok let's change the types when we get a chance.

File: Framework/include/Utility.h (hunk 3 above)
Comment (cmadrid1) [reply Framework/include/Utility.h:29]: It's ok to leave it. I was just curious. Good template function practice.

File: Framework/include/ScaleFactors.h (hunk 2 above)
Comment (JHiltbrand) [reply Framework/include/ScaleFactors.h:558]: Does it make sense that the above code just works without specifying `float` ?

File: Framework/include/Utility.h (hunk 3 above)
Comment (JHiltbrand) [reply Framework/include/Utility.h:33]: Turns out this method is a bit more useful now as I need to calculated the MT between a muon and MET for the Top tagger semileptonic selection and the `TLorentzVector` method simply does not work in this specific case.

**Pull request merged**